modules/nextflow/src/main/groovy/nextflow/processor/TaskConfig.groovy

```diff
@@ -358,7 +359,15 @@ class TaskConfig extends LazyMap implements Cloneable {
         if( !path )
             return null
 
-        return (path as Path).complete()
+        Path storeDir;
+        if (path instanceof Path)
+            storeDir = path
+        else if (path instanceof String)
```

Comment:
its actually pretty common: it comes up from patterns like 
```groovy
storeDir "a/b/c"
```

instead of 
```groovy
storeDir path("a/b/c")
```

before it was just a cast but that breaks LatchPaths bc it doesnt understand the latch scheme weirdly